algopy_testing/tests/arc4/test_uintn.py

```diff
@@ -0,0 +1,238 @@
+        (2**256, None),
+        (MAX_UINT512, None),
+    ],
+)
+def test_biguintn_overflow(get_avm_result: AVMInvoker, value: int, expected: int | None) -> None:
+    if expected is None:
+        with pytest.raises(algokit_utils.LogicError, match="assert failed"):
+            get_avm_result("verify_biguintn_init", a=int_to_bytes(value))
+        with pytest.raises(ValueError, match=f"expected value <= {2**256 - 1}"):
+            arc4.UInt256(value)
+    else:
+        avm_result = get_avm_result("verify_biguintn_init", a=int_to_bytes(value))
+        result = arc4.UInt256(value)
+        assert avm_result == expected
+        assert avm_result == result
+
+
+# where input bytes is not exactly 4 bytes, AVM returns None and
+# the stub implementation returns the expected value.
+# e.g. int_to_bytes(255, 2)
+# for the above test case, where input value is only 2 bytes,
+# AVM returns None and the stub implementation returns 255.
+@pytest.mark.parametrize(
+    "value",
+    [
+        int_to_bytes(0, 4),
+        int_to_bytes(255, 4),
+        int_to_bytes(2**16, 4),
+        int_to_bytes(2**32 - 1, 4),
+        int_to_bytes(2**32 - 1),
+    ],
+)
+def test_uintn_from_bytes(get_avm_result: AVMInvoker, value: bytes) -> None:
+    avm_result = get_avm_result("verify_uintn_from_bytes", a=value)
+    result = arc4.UInt32.from_bytes(value)
+    assert avm_result == result
+
+
+# where input bytes is not exactly 32 bytes, AVM returns None and
+# the stub implementation returns the expected value.
+# e.g. int_to_bytes(255, 4)
+# for the above test case, where input value is only 4 bytes,
+# AVM returns None and the stub implementation returns 255
+@pytest.mark.parametrize(
+    "value",
+    [
+        int_to_bytes(0, 32),
+        int_to_bytes(255, 32),
+        int_to_bytes(2**256 - 1, 32),
+        int_to_bytes(2**256 - 1),
+    ],
+)
+def test_biguintn_from_bytes(get_avm_result: AVMInvoker, value: bytes) -> None:
+    avm_result = get_avm_result("verify_biguintn_from_bytes", a=value)
+    result = arc4.UInt256.from_bytes(value)
+    assert avm_result == result
+
+
+# where input bytes is not exactly 4 bytes, AVM returns None and
+# the stub implementation returns the expected value.
+# e.g. (int_to_bytes(255, 2), ARC4_RETURN_PREFIX, 255)
+# for the above test case, where input value is only 2 bytes,
+# AVM returns None and the stub implementation returns 255.
+@pytest.mark.parametrize(
+    ("value", "prefix", "expected"),
+    [
+        (int_to_bytes(0, 4), ARC4_RETURN_PREFIX, 0),
+        (int_to_bytes(255, 4), ARC4_RETURN_PREFIX, 255),
+        (int_to_bytes(2**16, 4), ARC4_RETURN_PREFIX, 2**16),
+        (int_to_bytes(2**32 - 1, 4), ARC4_RETURN_PREFIX, 2**32 - 1),
+        (int_to_bytes(255, 4), b"", None),
+        (int_to_bytes(255, 4), b"\xff\x00\x01\x02", None),
+    ],
+)
+def test_uintn_from_log(
+    get_avm_result: AVMInvoker, value: bytes, prefix: bytes, expected: int | None
+) -> None:
+    if expected is None:
```

Comment:
Seperate this into two tests? one for the error scenarios and one for the happy path?

Reply:
They have been split into separate test cases as discussed in this commit 1251f4eb4e16e680d4315b3f30e50b2d98640e0c.